server/src/main/scala/org/scalaexercises/evaluator/evaluation.scala

```diff
@@ -463,14 +463,21 @@ class ${className} extends (() => Any) with java.io.Serializable {
     val currentClassPath = classPath.head
 
     // if there's just one thing in the classpath, and it's a jar, assume an executable jar.
-    currentClassPath ::: (if (currentClassPath.size == 1 && currentClassPath(0)
+    currentClassPath ::: (if (currentClassPath.size == 1 && currentClassPath.head
                               .endsWith(".jar")) {
-                            val jarFile = currentClassPath(0)
+                            val jarFile = currentClassPath.head
                             val relativeRoot =
-                              new File(jarFile).getParentFile()
-                            val nestedClassPath =
-                              new JarFile(jarFile).getManifest.getMainAttributes
-                                .getValue("Class-Path")
+                              new File(jarFile).getParentFile
+                            val nestedClassPath = Try {
+                              val jar = new JarFile(jarFile)
+                              val CP  = jar.getManifest.getMainAttributes.getValue("Class-Path")
+                              jar.close()
+                              CP
+                            } match {
+                              case Success(classPath) => classPath
+                              case Failure(throwable) =>
+                                throw new CompilerException(List(List(throwable.getMessage)))
+                            }
```

Comment:
Taking into account that we're not implementing this with effects, I'd go for a safer code:

```scala
  val nestedClassPath: String = try {
    val jar = new JarFile("")
    try {
      jar.getManifest.getMainAttributes.getValue("Class-Path")
    } finally {
      try {
        jar.close()
      } catch {
        case NonFatal(_) =>
      }
    }
  } catch {
    case NonFatal(e) =>
      throw new CompilerException(List(Option(e.getMessage).toList))
  }
```